### PR TITLE
[#156538443] - removed sendgrid configuration and unused argument in local provisioner

### PR DESCRIPTION
**Pull request by gunzip**
As we migrated to MailUp in production I've removed any sendgrid reference.

Aside, I've removed an unused variable (azurerm_functionapp) inside a local provisioner (sorry..)

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * ⚙️ [#156538443](https://www.pivotaltracker.com/story/show/156538443): Rimuovere la configurazione di sendgrid da Terraform

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: infrastructure/azure.tf [hunk 1]
```diff
@@ -663,7 +649,6 @@ resource "null_resource" "azurerm_app_service_portal" {
       "ts-node ${var.app_service_portal_provisioner}",
       "--environment ${var.environment}",
       "--azurerm_resource_group ${azurerm_resource_group.azurerm_resource_group.name}",
-      "--azurerm_functionapp ${azurerm_function_app.azurerm_function_app.name}",
```
Comment (cloudify) [inline infrastructure/azure.tf]: is this related to sendgrid?

Comment (gunzip) [reply infrastructure/azure.tf]: it isn't (ref. https://github.com/teamdigitale/digital-citizenship/pull/77#issue-179668607)

Comment (cloudify) [reply infrastructure/azure.tf]: then change the name of this PR

Comment (gunzip) [reply infrastructure/azure.tf]: done.
